backend/app/main/views.py

```diff
@@ -59,125 +85,333 @@ def get_mentor_by_email(email):
-            client.append(m.serialize())
-            return jsonify(client)
-    else:
-        return "This client does not exist in the database."
+    # Convert to JSON
+    response_json = response.json()
+
+    # Validation check
+    if (response.status_code // 100) != 2:
+        return response_json["error"], response.status_code
+    
+    # Get object's parameters
+    id_number = response_json["id"]
+    name = response_json["fields"].get("Name")
+    notes = response_json["fields"].get("Notes")
+    email = response_json["fields"].get("Client Email")
+    attachments = response_json["fields"].get("Attachments")
+    if name is None or email is None:
+        return "There is no client with this id. Please try again.", 422
 
+    # Create and return object
+    c = Client(name=name, email=email, id_number=id_number, notes=notes, attachments=attachments)
+    return jsonify(c.serialize()), 200
 
-# get a client from Airtable using client's email 
-@main.route("/clients/email/<email>", methods = ["GET"])
+
+# Get a client from Airtable using client's email 
 def get_a_client_from_email(email): 
     response = requests.get(
         "https://api.airtable.com/v0/appw4RRMDig1g2PFI/Clients?filterByFormula=SEARCH('{}'".format(email) + ", {Client Email})", 
         headers={"Authorization": str(os.environ.get("API_KEY"))},
     )
+    # Convert to JSON
     response_json = response.json()
 
-    list_of_clients = []
-    for r in response_json["records"]:
-        name = r["fields"].get("Name")
-        notes = r["fields"].get("Notes")
-        attachments = r["fields"].get("Attachments")
-        if name is not None:
-            m = Client(name=name, notes=notes, attachments=attachments)
-            list_of_clients.append(m.serialize())
-            return jsonify(list_of_clients)
-    else: 
-        return "There is no client with that email, please try again."
+    # Validation check
+    if (response.status_code // 100) != 2:
+        return response_json["error"], response.status_code
+    
+    error = handleEmailResponse(response_json["records"])
+    if(error is not None):
+        return error, 422
+    
+    # Get object's parameters
+    id_number = response_json["records"][0]["id"]
+    name = response_json["records"][0]["fields"].get("Name")
+    email = response_json["records"][0]["fields"].get("Client Email")
+    notes = response_json["records"][0]["fields"].get("Notes")
+    attachments = response_json["records"][0]["fields"].get("Attachments")
+    if name is None or email is None:
+        return "There is no client with that email. Please try again.", 422
+    
+    # Create and return object
+    c = Client(name=name, email=email, id_number=id_number, notes=notes, attachments=attachments)
+    return jsonify(c.serialize()), 200
 
-# get all Volunteers from Airtable
-@main.route("/volunteers", methods = ["GET"])
+# VOLUNTEER ROUTE
+@main.route("/volunteers/", methods=["GET"])
```

Comment:
I have two routes for each class. This is the way to have both paths route to the same function so that if someone enters either /volunteers or /volunteers/, he would still get the list of all volunteers. It is the same for mentors, clients and donors.
Of course, we can change it.